plugins/modules/controller.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, Network to Code (@networktocode) <info@networktocode.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# TODO: Check for any words of Device
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: controller
+short_description: Create, update or delete controllers within Nautobot
+description:
+  - Creates, updates or removes controllers from Nautobot, related page: https://docs.nautobot.com/projects/core/en/stable/user-guide/core-data-model/dcim/controller/
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Josh VanDeraa (@jvanderaa)
+version_added: "5.3.0"
+extends_documentation_fragment:
+  - networktocode.nautobot.fragments.base
+  - networktocode.nautobot.fragments.tags
+  - networktocode.nautobot.fragments.custom_fields
+options:
+  name:
+    description:
+      - The name of the controller
+    required: true
+    type: str
+    version_added: "5.3.0"
```

Comment:
FYI, these are not needed for every option unless they differ from the top level `version_added`. It doesn't hurt necessarily, but it is redundant and not required.

See reference under [Documentation fields > options > version_added](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fields):

> Only needed if this option was extended after initial Ansible release, in other words, this is greater than the top level version_added field.